### ISSUE TITLE
Layout idea : Session measure bar on Chord progression display.

### ISSUE THREAD

**Issue by NakajimaYusuke** [elieserdejesus/JamTaba#206]
Hi, I'm looking forward to the update of jamtaba every day.
This post, I have an idea for the chord display layout.

As everyone knows, jamtaba can display the "chord progression".
And jamtaba can also display "session measure bar". (Aka : Session time bar or click bar. )

When these 2 are same size and next to each other(or these are overlapped), I think it would be convenient to see.
Please see this mock up picture.

![timebaridea1](https://cloud.githubusercontent.com/assets/15982547/11918693/38abc5b6-a77f-11e5-89f1-7de2b5f8e9e5.png)
When I chose "line" as measure bar shape.
I want "each click (each number)" to run through "under chord progress display".
By doing so , we can see the "chord progression display" and "measure bar" at the same time and same place.

Even at present, we can also read "information about the time" from the "chord progression display".
In other words, "chord progression display" plays the role of "simplified session measure bar".
Therefore, It should be useful when "chord progression display" and "session measure bar" are displayed side by side.

![timebaridea2](https://cloud.githubusercontent.com/assets/15982547/11918691/26c99a80-a77f-11e5-8d18-08d955d11c66.png)

This mock-up picture is another idea.
There is a chord proggression display on the measure bar.

Anyway, I prefer the display style that they are not independent.
When it can be indicated together, it'll be nice. They should be same size and stick together!

Thank you.

**Comment by elieserdejesus** [comment]
Hi Nakajima, I agree with your idea. Actually in the first Jamtaba version I used exactly this idea: [https://www.youtube.com/watch?v=rzwyUkrBKNM](https://www.youtube.com/watch?v=rzwyUkrBKNM)

I have some issues to discuss before change the current layout:

- when user is using "Line" as interval progress shape you see the "line" with the interval beats. It's ok. Now, if you accept a chord progression you see the chords. If we use your first suggestion the interval beats will be drawed twice: in the "original place" and again in the "chords area"?

- We can draw the interval beats in "chords area" **only** if user is playing some chord progression. If not the interval beats are drawed in the "original" place. But if we draw the interval beats in "chords area" the interval beats "original area" will be empty?

- I'm not shure if we need the beat numbers in chords. I think we need just some feedback about the interval progression in the chords area. At moment we know when the **chords change**, this is clear because the current chord is highlighted. But is not clear when the **beat change**. 

- **is enough see the beat changing in chords area?**

**Comment by NakajimaYusuke** [comment]
Thank you for your reply.

I tried the previous version of jamtaba.
It's chord display style is exactly the format that I desire!

Of course, I think it is sufficiently convenient even in the chord display style of the current jamtaba2.
Therefore, the priority of this suggestion wouldn't be so high.
However, if I compare two, I prefer the style of the previous version.

>•when user is using "Line" as interval progress shape you see the "line" with the interval beats. It's ok. Now, if you accept a chord progression you see the chords. If we use your first suggestion the interval beats will be drawed twice: in the "original place" and again in the "chords area"?


>•We can draw the interval beats in "chords area" only if user is playing some chord progression. If not the interval beats are drawed in the "original" place. But if we draw the interval beats in "chords area" the interval beats "original area" will be empty?

I recommend my first suggestion's style.
When user is using "Line" as interval progress shape, each click run through on the chord display.
(In this case, the beat number of each clicks may unnecessarily...)

Then, the interval beats "original area" will be empty.
I think that jamtaba GUI do not need to draw "interval beats" twice.
Interval beats is enough in one place.

However, it's a little wasteful for there to remain empty area.
At this, "occurred empty area" should be effectively used for another purpose.
I recommend the built-in chords book in this empty area.
![oldjamtabachords1](https://cloud.githubusercontent.com/assets/15982547/12082789/99d24ac4-b2db-11e5-9aad-16ca2f3f2db1.png)
There was such function in previous version of jamtaba.
Although it was a very simplified one, I think it's convenient.



Please see this mock-up image. (Sorry for the GUI of mingled with Japanese.)
![chordbook1](https://cloud.githubusercontent.com/assets/15982547/12082711/22590178-b2da-11e5-923f-770b266a592a.png)

I embedded a "chords book sofrware" to the "site of the interval beats (empty area)" in this image.
If user can save and load each chord progressions in this area, That will be very convenient.

　


For your information, this chords book was made by our hero tea.
SongBookUtil
https://code.google.com/p/pyninjam/wiki/SongBookUtil

I'm sorry if you already know it.
This is a useful tool for song presented during the ninjam session.
When I load the saved song, It does "pasting chord progression" and "changing BPM and BPI" at the same time.
If such tool is embedded in a ninjam client, "users who prefer the complex chord progression" will rejoice.

**Comment by elieserdejesus** [comment]
Nakajima, thanks fot the very nice ideas with a lot of details. 
I'm not happy enough with the current visual aspect of the chord progression display. It's working, but can be better; Shure I will use your ideas. But the "song selection" is more complex, but this is a old idea I have too (as you can see in the first Jamtaba).
Let's solve the layout first, and work in a chord progression database in the future.

I like this idea: **When I load the saved song, It does "pasting chord progression" and "changing BPM and BPI" at the same time**.
I always think in a "chord progression database", but now you is proposing some very interesting; a "song database". A chord progression contain just chords, but a song contain BPM and BPI. Off course the 2 things are very usefull and we can do both. 
   Thanks for the ideas! @teamikl , maybe we can start planning something for a song/chord progression database?

**Comment by elieserdejesus** [comment]
Hi @NakajimaYusuke, I'm improving the chords display to show beats progression in chords. I keep the "beat circles", but when you are using chords the "beat circles" are a bit darker. The idea is "keep the focus in chords when using chords".  I changed the color of highlighted colors too.

I will close this issue but I'm creating a new issue related to the progressions/song database.

![capture](https://cloud.githubusercontent.com/assets/1012741/12279223/2388bd7e-b96d-11e5-82d4-477a292b0e05.gif)

**Comment by NakajimaYusuke** [comment]
How wonderful! This visibility is perfect.
The chord progression display also got beats display with this.

I'm looking forward to the next update. Thank you for fulfill my dream!

**Comment by elieserdejesus** [comment]
Nakajima, you're welcome, thanks for the nice ideas!
